### PR TITLE
:bug: Adding use of new Maven writeSettings to fully write the maven settings

### PR DESCRIPTION
**Pull request by shawn-hurley**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -110,3 +110,5 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace github.com/konveyor/tackle2-addon => /Users/shurley/repos/tackle2-addon
```
Comment (jortel) [inline go.mod:114]: Reminder to remove this before merged. ^^

File: cmd/mode.go [hunk 2]
```diff
@@ -29,32 +29,39 @@ type Mode struct {
...
 func (r *Mode) Build(application *api.Application) (err error) {
+	binDir := path.Join(BinDir, "maven")
+	maven := &repository.Maven{
+		M2Dir:  M2Dir,
+		BinDir: binDir,
+		Remote: repository.Remote{
+			Identities: application.Identities,
+		},
+	}
+	path, err := maven.writeSettingsFile()
```
Comment (jortel) [inline cmd/mode.go:42]: Shouldn't this be maven.CreateSettingsFile()
I don't think this method exists and isn't exported.  How does this compile?

File: cmd/mode.go [hunk 3]
```diff
@@ -119,17 +123,8 @@ func (r *Mode) getArtifact() (err error) {
...
-func (r *Mode) mavenArtifact(application *api.Application) (err error) {
-	binDir := path.Join(BinDir, "maven")
-	maven := repository.Maven{
-		M2Dir:  M2Dir,
-		BinDir: binDir,
-		Remote: repository.Remote{
-			Identities: application.Identities,
-		},
-	}
+func (r *Mode) mavenArtifact(application *api.Application, maven *repository.Maven, binDir string) (err error) {
```
Comment (jortel) [inline cmd/mode.go:127]: I don't think `binDir` is used anymore.

Comment (jortel) [inline cmd/mode.go:127]: Can't the code below use maven.BinDir instead of passing separately?

File: go.mod (hunk 1 above)
Comment (shawn-hurley) [reply go.mod]: updated now that the add-on PR is merged!

File: cmd/mode.go (hunk 2 above)
Comment (shawn-hurley) [reply cmd/mode.go]: This should be updated; this should never have worked my bad

File: cmd/mode.go (hunk 3 above)
Comment (shawn-hurley) [reply cmd/mode.go]: Updated that is much cleaner!

**Comment by shawn-hurley** [comment]
@jortel @mansam I think that this one is ready for review. 

Sorry about the removal of the `//`

**Review by jortel — APPROVED** [review]
LGTM

**Pull request merged**